webapp-client/src/component/AddCategoryForm.js

```diff
@@ -1,32 +1,76 @@
 import React from 'react';
+import ErrorMessage from './ErrorMessage';
 
 export default class AddCategoryForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+    this.createCategory = this.createCategory.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
+  }
   // This function will be changed later to directly POST request to db instead of calling addCategory
-  createCategory = (e) => {
+  async createCategory(e) {
     e.preventDefault();
     const category = e.target.elements['add-category'].value;
-    this.props.addCategory(category);
-    e.target.elements['add-category'].value = ''; 
-  };
+    e.target.elements['add-category'].value = '';
+    const request = {
+      method: 'POST',
+      body: JSON.stringify({
+        title: category,
+        userId: 'fake-user-id',
+        items: [],
+      }),
+      headers: { 'Content-type': 'application/json' },
+    };
+    try {
+      const response = await fetch(
+        'http://localhost:8080/category/createCategory',
+        request
+      );
+      if (response.status !== 200) {
+        throw new Error(response.statusMessage);
+      }
+      // to be deleted once CategoryList fetches from db
+      this.props.addCategory(category);
+    } catch (error) {
+      console.log(error);
```

Comment:
delete maybe?

Reply:
Ah currently I log all of the errors but is that considered debugging message?

Reply:
discussed:)